src/carnot/funcs/builtins/pprof.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2018- The Pixie Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <sys/sysinfo.h>
+
+#include <absl/container/flat_hash_map.h>
+
+#include <string>
+
+#include "src/shared/pprof/pprof.h"
+
+#include "src/carnot/udf/registry.h"
+#include "src/carnot/udf/udf.h"
+
+namespace px {
+namespace carnot {
+namespace builtins {
+
+using px::shared::PProfProfile;
+
+class CreatePProfRowAggregate : public udf::UDA {
+ public:
+  static udf::UDADocBuilder Doc() {
+    return udf::UDADocBuilder("Convert perf profiling data to pprof format.")
+        .Details("Converts perf profiling stack traces into pprof format.")
+        .Example("df = df.agg(pprof=('stack_trace', 'count', px.pprof))")
+        .Arg("stack_trace", "Stack trace string.")
+        .Arg("count", "Count of the stack trace string.")
+        .Returns("A single row that aggregates all the stack traces and counts into pprof format.");
+  }
+
+  void Update(FunctionContext*, const StringValue stack_trace, const Int64Value count) {
+    histo_[stack_trace] += count.val;
+  }
+
+  void Merge(FunctionContext*, const CreatePProfRowAggregate& other) {
+    for (const auto& [stack_trace, count] : other.histo_) {
+      histo_[stack_trace] += count;
+    }
+  }
+
+  StringValue Serialize(FunctionContext*) {
+    static constexpr uint32_t stack_trace_period_ms = 11;
```

Comment:
Should this be `FLAGS_stirling_profiler_stack_trace_sample_period_ms` rather than being hard coded?

Reply:
Currently keeping this as hard coded. Added issue #1352. Open to further discussion.